### PR TITLE
Add document for cleaning up commits in a pull request

### PR DESCRIPTION
**Pull request by cormacpayne**
Also changed pull request template to include a section for comments at the top, include a note about removing unused check boxes, and include a link to the commit document.

Fix for issue https://github.com/Azure/azure-powershell/issues/2843

---

This checklist is used to make sure that common issues in a pull request are covered by the creator. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270%28v=vs.85%29.aspx).

Below in **Overall Changes**, check off the boxes that apply to your PR. Within each of the categories that you select, make sure that you can check off **all** of the boxes.
## Overall Changes
- [x] [**MANDATORY** - General changes](#general)
### General
- [x] Title of the PR is clear and informative
- [x] There are a small number of commits that each have an informative message
- [x] If it applies, references the bug/issue that the PR fixes
- [x] All files have the Microsoft copyright header
- [x] Cmdlets refer to management libraries through nuget references - no dlls are checked in
- [x] The PR does not introduce breaking changes (unless a major version change occurs in the assembly and module)

### CODE REVIEW

File: documentation/cleaning-up-commits.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+### Number of commits
+
+It can be difficult to follow the changes in a pull request when the number of commits that come with it become too large:
+- If a bug fix is being addressed, a single commit should be submitted
+- If a new feature is being introduced, then the pull request can have multiple logical commits with each commit clearly describing what it does
+
+### Rebasing
+
+Sometimes a pull request can be based on a much earlier commit in the branch that you are trying to merge into it, causing a large amount of commits and file changes to litter the pull request. In this case, it would be better to **rebase** (move branches around by changing the commit that they are based on). After rebasing, you will want to close the pull request and open a new one, which will now have less commits.
```
Comment (markcowl) [inline documentation/cleaning-up-commits.md]: <nit>less commits -> fewer commits </nit>

File: documentation/cleaning-up-commits.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+For example, if you're working off of the branch **feature** and are trying to rebase with **master**, you may run one of the following commands:
```
Comment (markcowl) [inline documentation/cleaning-up-commits.md]: off of -> from

**Comment by markcowl** [comment]
@cormacpayne a couple of nits, this looks really good.

**Comment by cormacpayne** [comment]
@markcowl resolved review comments

**Comment by cormacpayne** [comment]
@azuresdkci test this please

**Pull request merged**